pennylane/interfaces/batch/autograd.py

```diff
@@ -0,0 +1,170 @@
+        execute_fn=execute_fn,
+        gradient_fn=gradient_fn,
+        _n=_n,
+    )[0]
+
+
+@autograd.extend.primitive
+def _execute(
+    parameters, tapes=None, device=None, execute_fn=None, gradient_fn=None, _n=1
+):  # pylint: disable=dangerous-default-value,unused-argument
+    """Autodifferentiable wrapper around ``Device.batch_execute``.
+
+    The signature of this function is designed to workaround Autograd restrictions.
+    Note that the ``parameters`` argument is dependent on the ``tapes`` argument;
+    this function should always be called as follows:
+
+    >>> parameters = [autograd.builtins.list(t.get_parameters()) for t in tapes])
+    >>> parameters = autograd.builtins.tuple(parameters)
+    >>> _batch_execute(parameters, tapes=tapes, device=device)
+
+    In particular:
+
+    - ``parameters`` is dependent on the provided tapes: always extract them as above
+    - ``tapes`` is a *required* argument
+    - ``device`` is a *required* argument
+
+    The private argument ``_n`` is used to track nesting of derivatives, for example
+    if the nth-order derivative is requested. Do not set this argument unless you
+    understand the consequences!
+    """
+    with qml.tape.Unwrap(*tapes):
+        res, jacs = execute_fn(tapes)
+
+    return [np.tensor(r) for r in res], jacs
+
+
+def vjp(
+    ans, parameters, tapes=None, device=None, execute_fn=None, gradient_fn=None, _n=1
+):  # pylint: disable=dangerous-default-value,unused-argument
+    """Returns the vector-Jacobian product operator for a batch of quantum tapes.
+
+    Args:
+        ans (array): the result of the batch tape execution
+        parameters (list[list[Any]]): Nested list of the quantum tape parameters.
+            This argument should be generated from the provided list of tapes.
+        tapes (Sequence[.QuantumTape]): batch of tapes to execute
+        device (.Device): Device to use to execute the batch of tapes.
+            If the device does not provide a ``batch_execute`` method,
+            by default the tapes will be executed in serial.
+        execute_fn (callable): The execution function used to execute the tapes
+            during the forward pass. This function must return a tuple ``(results, jacobians)``.
+            If ``jacobians`` is an empty list, then ``gradient_fn`` is used to
+            compute the gradients during the backwards pass.
+        gradient_fn (callable): the gradient function to use to compute quantum gradients
+        _n (int): a positive integer used to track nesting of derivatives, for example
+            if the nth-order derivative is requested.
+
+    Returns:
+        function: this function accepts the backpropagation
+        gradient output vector, and computes the vector-Jacobian product
+    """
+
+    def grad_fn(dy):
+        """Returns the vector-Jacobian product with given
+        parameter values p and output gradient dy"""
+
+        dy = dy[0]
+        jacs = ans[1]
+
+        if jacs:

```

Comment:
`jacs` will always be a sequence of Jacobians, each one corresponding to an input circuit, so it will never be `0` (unless something has gone very wrong on the device).

And because `jacs` is always `type(list)`, we need to follow PEP8, which enforces

![image](https://user-images.githubusercontent.com/2959003/128522107-42a97c25-392a-4dc2-b3c3-e26925ebcf54.png)

Reply:
That's great that PEP8 recommends it, but I'm always wary of this pattern because people end up using it also in the case where the type can be one of several options, and it has caused bugs in the past

Reply:
Yep. I think that one that always occurs is when something can be `None` _or_ `[]`. In this case, `if thing:` is ambiguous.